espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/RouteDefinition.java

```diff
@@ -0,0 +1,76 @@
+package io.appium.espressoserver.lib.http;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import fi.iki.elonen.NanoHTTPD.Method;
+import io.appium.espressoserver.lib.handlers.RequestHandler;
+import io.appium.espressoserver.lib.model.AppiumParams;
+
+class RouteDefinition {
+
+    private final String testRegex;
+    private final String routeUri;
+    private final Method method;
+    private final Class<? extends AppiumParams> paramClass;
+    private final RequestHandler<? extends AppiumParams, ?> handler;
+
+    RouteDefinition(Method method, String routeUri, RequestHandler<? extends AppiumParams, ?> handler, Class<? extends AppiumParams> paramClass) {
+        testRegex = buildTestRegex(routeUri);
+        this.routeUri = routeUri;
+        this.method = method;
+        this.paramClass = paramClass;
+        this.handler = handler;
+    }
+
+    public boolean isMatch (String uri) {
+        return uri.matches(testRegex);
+    }
+
+    public Map<String, String> getUriParams(String uri) {
```

Comment:
this can be also done with Apache utils:

```
import org.apache.hc.client5.http.utils.URLEncodedUtils

String url = "http://www.example.com/something.html?one=1&two=2&three=3&three=3a";
List<NameValuePair> params = URLEncodedUtils.parse(new URI(url), "UTF-8");

for (NameValuePair param : params) {
  System.out.println(param.getName() + " : " + param.getValue());
}
```